### DISCUSSION TITLE
规则问题，请求指导

### DISCUSSION THREAD

**Discussion by liumingan** [nikkinikki-org/OpenWrt-nikki#96] [Q&A]
我想建立一个把所有cloud flare的IP都转发到指定的优选域名的规则，不知道怎么弄，请知道的指点

**Comment by zutzo** [comment]
如果你走代理 请忽略 
如果你想直连 请使用 https://github.com/sbwml/luci-app-mosdns 插件 作为 clash 的上游 并 不要使用 FakeIP

**Reply by liumingan** [reply]
我想通过mihomo实现，不想再使用其他的DNS转发

**Reply by zutzo** [reply]
这不属于 mihomo 范围 
最多只能单个域名写 hosts

**Reply by liumingan** [reply]
我觉得应该可以实现，只是我不会。
假如优选的域名是：a.com
cloudflare的IP是 ：ANS，13335

或者是不能实现？

**Reply by zutzo** [reply]
走代理还用优选 IP/域名 的意义是?

**Reply by liumingan** [reply]
走优选就不再走代理了，用cloud flare的CDN

**Reply by zutzo** [reply]
那你又不肯另装插件 
mihomo 不支持 这么弄
mihomo 是 代理 工具 不是 DNS 转发器

**Comment by morytyann** [comment]
使用`geoip:cloudflare`即可，详见[此说明](https://github.com/MetaCubeX/meta-rules-dat?tab=readme-ov-file#countrymmdbgeoipdatgeoipdb-%E5%86%85%E5%AE%B9)

**Reply by morytyann** [reply]
这个是配置在配置文件的rules下的，详见[官方文档](https://wiki.metacubex.one/config/rules/)，你说的IP-ASN，mihomo也支持这个规则。

**Comment by liumingan** [comment]
问题是我不知道应该怎么转发给优选域名
GOEIP,cloudflare,******
这个规则怎么写呢，不能用直连或者代理吧


Joseph Mory ***@***.***> 于 2024年8月21日周三 09:15写道：

**Comment by morytyann** [comment]
哦~这个优选域名的规则是指DNS解析的规则？你可以试试[nameserver-policy](https://wiki.metacubex.one/config/dns/#nameserver-policy)？但是如果你是想解析出cloudflare的IP后再转发到优选规则，这个目前应该做不到。

**Reply by liumingan** [reply]
我好像见到过有人用tunnels转发的配置，可是不知道怎么调用这个tunnels
tunnels:
  - [tcp,udp],1270.0.1:1234,a.com

**Reply by morytyann** [reply]
可以多看看Mihomo的[官方文档](https://wiki.metacubex.one/config/tunnels/)。

**Reply by liumingan** [reply]
解析出是cloud flare的IP就通过隧道转发，隧道可以经过也可以不经过代理，问题是怎么调用这个隧道呢

**Reply by morytyann** [reply]
我不太清楚，没有相关的需求，你试试去Mihomo仓库或者相关的论坛问下，这里人太少了，哈哈。
